src/components/accession2/Accession2View.tsx

```diff
@@ -112,7 +120,11 @@ export default function Accession2View(props: Accession2ViewProps): JSX.Element
             <IconButton sx={{ marginLeft: 3, height: '24px' }} onClick={() => setOpenDeleteAccession(true)}>
               <Icon name='iconTrashCan' />
             </IconButton>
-            <Button onClick={() => setOpenWithdrawModal(true)} label={strings.WITHDRAW} />
+            {accession && accession.state === 'Awaiting Check-In' ? (
```

Comment:
we have` if (accession) ` code in this file.. this component does not make sense without an accession. it should not be an optional property.

Reply:
The problem here is that we get the accession from the accession id in the url, and if the id doesn't exist, then the accession is null/undefined

Reply:
> The problem here is that we get the accession from the accession id in the url, and if the id doesn't exist, then the accession is null/undefined

We should then make the accession view a separate component, and the code that looks for accession id a parent component. This is the ideal way to design here.

Reply:
can we at least follow up separately? we don't need to put all the code in one component, we should separate by intent when we can.

Reply:
Yes, of course